src/backend/distributed/planner/deparse_shard_query.c

```diff
@@ -105,13 +106,13 @@ RebuildQueryStrings(Query *originalQuery, List *taskList)
 		}
 
 		ereport(DEBUG4, (errmsg("query before rebuilding: %s",
-								task->queryString == NULL ? "(null)" :
-								ApplyLogRedaction(task->queryString))));
+								TaskQueryString(task) == NULL ? "(null)" :
+								ApplyLogRedaction(TaskQueryString(task)))));
 
 		UpdateTaskQueryString(query, relationId, valuesRTE, task);
```

Comment:
Is it possible to defer query string generation here as well? This is the code-path called for all INSERTs, so we're not lazily deferring for INSERTs?

Reply:
Should be resolved in the latest commit